src/libraries/sendtohelix.proj

```diff
@@ -1,122 +1,172 @@
+    "correlation payload", which is the set of files used by all Helix submissions
+    (which we compress into a single file).
+-->
 
-    <!-- This property is used to show the tests results in Azure Dev Ops. By setting this property the
-    test run name will be displayed as $(BuildSettings)-$(HelixTargetQueue) -->
-    <TestRunNamePrefix>$(BuildSettings)-</TestRunNamePrefix>
-    <TestRunNamePrefix Condition="'$(TestRunNamePrefixSuffix)' != ''">$(TestRunNamePrefix)$(TestRunNamePrefixSuffix)-</TestRunNamePrefix>
+<Project DefaultTargets="RunInParallelForEachScenario">
 
-    <FailOnTestFailure Condition="'$(WaitForWorkItemCompletion)' != ''">$(WaitForWorkItemCompletion)</FailOnTestFailure>
-  </PropertyGroup>
+  <!-- Helix auto-magically imports local *.props files. We're not going to import the Helix Sdk,
+       so we need to import those files manually so we can reference the properties defined there.
+  -->
+  <Import Project="$(MSBuildThisFileDirectory)/*.props" />
+
+  <Target Name="RunInParallelForEachScenario">
+
+    <PropertyGroup>
+      <!-- This specifies what properties are needed to be passed down as global properties to a child project invocation. -->
+
+      <_PropertiesToPass>
+        RuntimeFlavor=$(RuntimeFlavor);
+        TargetArchitecture=$(TargetArchitecture);
+        Configuration=$(Configuration);
+        TargetOS=$(TargetOS);
+        TestRunNamePrefixSuffix=$(TestRunNamePrefixSuffix);
+        Creator=$(Creator);
+        HelixAccessToken=$(HelixAccessToken);
+        HelixTargetQueues=$(HelixTargetQueues);
+        BuildTargetFramework=$(BuildTargetFramework);
+        BuildSettings=$(BuildSettings)
+      </_PropertiesToPass>
+    </PropertyGroup>
+
+    <Message Condition="'$(_Scenarios)' != ''" Importance="High" Text="Using _Scenarios: $(_Scenarios)" />
+    <Message Importance="High" Text="Using Queues: $(HelixTargetQueues)" />
+    <Message Importance="High" Text="BuildSettings: $(BuildSettings)" />
+    <Message Importance="High" Text="BuildTargetFramework: $(BuildTargetFramework)" />
+    <Message Importance="High" Text="TestArchiveTestsRoot: $(TestArchiveTestsRoot)" />
+    <Message Importance="High" Text="TestHostRootPath: $(TestHostRootPath)" />
+    <Message Importance="High" Text="TestArchiveRoot: $(TestArchiveRoot)" />
+    <Message Importance="High" Text="TestArchiveRuntimeRoot: $(TestArchiveRuntimeRoot)" />
+    <Message Importance="High" Text="TestArchiveRuntimeFile: $(TestArchiveRuntimeFile)" />
 
-  <PropertyGroup Condition="'$(BuildAllConfigurations)' == 'true'">
-    <!-- Use helix feature to include dotnet CLI for every workitem and add it to the path -->
-    <IncludeDotNetCli>true</IncludeDotNetCli>
-    <DotNetCliPackageType>sdk</DotNetCliPackageType>
+    <!-- Re-invoke MSBuild on this project to create the correlation payload -->
+    <MSBuild Projects="$(MSBuildProjectFile)" Targets="PrepareCorrelationPayloadDirectory" Properties="Scenarios=$(_Scenarios)" />
```

Comment:
Do we need to re-invoke this project? We could just make the current target depend on PrepareCorrelationPayloadDirectory; we didn’t need to do this as an extra MSbuild node.